frontend/src/components/Network.tsx

```diff
@@ -379,6 +381,12 @@ const Network = ({ nodes: _nodes, edges: _edges }: Props) => {
     [_edges, nodes, minEdgeStrength, maxEdgeStrength, edgeColors],
   );
 
+  /** fit view to contents */
+  const fit = async () => {
+    await sleep(10);
```

Comment:
Why the sleep, out of curiosity? Just giving time for things to settle down?

Reply:
Yes, this is a little bit hand-wavey, and it's really only needed where I'm calling it from `graph.current.on("layoutstop", fit)` because I found that some of the layout algorithms (I think the ones that are more continuous/animated) aren't truly done when that event is called.

Reply:
I just pushed a change to clarify this.

Idk if it's the layout algo plugins that are lying about being truly done or if it's some shortcoming in the cytoscape core, and I didn't feel it's worth it to investigate further atm, but in testing this timeout value seems to be sufficient for now.